src/inventory.h

```diff
@@ -161,6 +161,22 @@ struct ItemStack
 	// Similar to takeItem, but keeps this ItemStack intact.
 	ItemStack peekItem(u32 peekcount) const;
 
+	bool operator ==(const ItemStack &s) const
+	{
+		if (this->name != s.name ||
```

Comment:
why not just return the condition directly? Also `==` and `&&` would seem more logical here

Reply:
Good idea :+1:

Reply:
Done. Code's now much simpler, thanks. :)